### PR TITLE
Added Go 1.5 base image

### PR DESCRIPTION
**Pull request by paddycarey**
This PR is a first attempt at defining a common base image to use across all ShopKeep Go services.

This image includes the most common Go tools (vet, lint, godep, etc) that we use across all of our Go applications and removes the need to handle installation of these on a per-application basis.

### CODE REVIEW

**Comment by declantraynor** [comment]
:+1:

**Comment by wchan2** [comment]
:+1:

File: go/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM golang:1.5.2
+MAINTAINER ShopKeep Developers <developers@shopkeep.com>
+
+RUN apt-get update && \
+    apt-get install -y build-essential && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+RUN go get github.com/tools/godep
```
Comment (pguelpa) [inline go/Dockerfile]: I think we can pass multiple packages to `go get` and it would reduce the number of layers in the image.

**Comment by pguelpa** [comment]
So a question about how useful we feel this will be vs. just having each service specify the version of Go in their `Dockerfile`. In my mind we wouldn't want to have all our services just use the `latest` tag of this because it would mean we would be breaking the ability to build services consistently.

**Comment by paddycarey** [comment]
I think it could be useful for several reasons:

- The base image would contain any shared setup that we know is vital to all our Go images (vet, lint, godep, build-essential) and save both build time and disk space by sharing those lower layers across all services.
- Tests would run faster as we wouldn't have to install those tools at runtime every time we run the tests like we do now.
- The version of the tools installed at image build time would be baked into our base image and would prevent issues like the Godep problem we had today (e.g. we know if we inherit from `shopkeep/go:1.5` that it will have `godep` installed and ready to use at a known version).

I wouldn't want to use the `latest` tag either, we could trigger different image tags based on git tags just like we do for `shpkpr`, so we could do something like `FROM shopkeep/golang:1.5.2`. There would be no change from how we do things currently, this is more about shared dependencies than anything else.

**Comment by pguelpa** [comment]
Cool, I like the idea @paddycarey :+1:

**Pull request merged**